velox/functions/prestosql/tests/ArithmeticTest.cpp

```diff
@@ -177,6 +177,42 @@ TEST_F(ArithmeticTest, ln) {
   EXPECT_EQ(std::nullopt, ln(std::nullopt));
 }
 
+TEST_F(ArithmeticTest, log2) {
+  const auto log2Eval = [&](std::optional<double> a) {
+    return evaluateOnce<double>("log2(c0)", a);
+  };
+
+  std::vector<double> values = {123, 123.45, 1, 65536, 0.125};
+  for (double value : values) {
+    EXPECT_EQ(std::log2(value), log2Eval(value));
+  }
+
+  values = {-123, -123.45};
+  for (double value : values) {
+    EXPECT_TRUE(std::isnan(log2Eval(value).value()));
+  }
+
+  EXPECT_EQ(std::nullopt, log2Eval(std::nullopt));
+}
+
+TEST_F(ArithmeticTest, log10) {
+  const auto log10Eval = [&](std::optional<double> a) {
+    return evaluateOnce<double>("log10(c0)", a);
+  };
+
+  std::vector<double> values = {123, 123.45, 1000, 1};
```

Comment:
I would only have a few hand-written and hand-crafted cases. It's easier to verify.

```
EXPECT_EQ(log10(kInf), kInf);
EXPECT_EQ(log10(10), 1);
EXPECT_EQ(log10(1), 0);
EXPECT_EQ(log10(.1), -1);
EXPECT_THAT(log10(-1), IsNan());
EXPECT_THAT(log10(-kInf), IsNan());
EXPECT_THAT(log10(kNan), IsNan());
EXPECT_THAT(log10(std::nullopt), std::nullopt);
```

Reply:
I was following the unit tests from here:
https://github.com/prestodb/presto/blob/5dd94fe7c70c98960c57782ef9f53fad3eb4218d/presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java#L459-L474

This suggests that presto does not throw exception for invalid values (e.g. negative).

jdk doc for log10:https://docs.oracle.com/javase/7/docs/api/java/lang/Math.html
cpp: https://en.cppreference.com/w/cpp/numeric/math/log10

In both cases, if argument is less than zero, result is NAN. Since Presto unit tests checks in math.log10, the results here and presto oss are identical.

Reply:
I'll defer to the Presto folks, but IMO writing the tests that way is a mistake. It should be quick and easy to verify that the key edge cases and error states (null, inf, -inf, nan, negative numbers) are covered and what the asserted behavior is.

Case in point: With the exception of `log10(NULL)`, I can't tell from that test file what the behavior for these edge cases should be. I'm just taking your word for it.

Reply:
I think Sagar has a good point.

Reply:
> I can't tell from that test file what the behavior for these edge cases should be. I'm just taking your word for it.

I actually did not understand what you are saying. Every expect statement takes input and has expectation, are you saying that is no clear?  E.g. `EXPECT_EQ(std::log2(value), log2Eval(value));` means we call velox log2 function with a variable value, and match it with std::log2. Not sure what is not clear here.

Reply:
Updated the tests as suggested by @funrollloops